sys/include/net/gcoap.h

```diff
@@ -602,6 +621,13 @@ extern "C" {
 #define GCOAP_RESEND_BUFS_MAX      (1)
 #endif
 
+/* define a common name for transport layer object, to accommodate security layer */
+#ifdef MODULE_SOCK_DTLS
+typedef sock_dtls_t coap_tl_sock_t;
+#else
+typedef sock_udp_t coap_tl_sock_t;
+#endif
```

Comment:
Doxygen docs complain about absent documentation for these definitions. Since they are used only in _listen() in `gcoap.c`, you can move these definitions there.

Reply:
Moved to `gcoap.c` and also added usage instruction to docs in c52b34b.